src/sentry/tasks/statistical_detectors.py

```diff
@@ -241,6 +307,78 @@ def _detect_transaction_trends(
     )
 
 
+def query_transactions_timeseries(
+    transactions: List[Tuple[int, int | str]],
+    start: datetime,
+    agg_function: str,
+) -> Generator[Tuple[int, Union[int, str], SnubaTSResult], None, None]:
+    end = start.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+    interval = 3600  # 1 hour
+
+    grouped_transactions = defaultdict(list)
+    for project_id, transaction_name in transactions:
+        grouped_transactions[project_id].append(transaction_name)
+
+    for project_id, transaction_names in transactions:
+        params: Dict[str, Any] = {
+            "start": end - timedelta(days=14),
+            "end": end,
+            "project_id": [project_id],
+            "project_objects": [Project.objects.get(id=project_id)],
+        }
+
+        # Snuba allows 10,000 data points per request. 14 days * 1hr * 24hr =
+        # 336 data points per transaction name, so we can safely get 25 transaction
+        # timeseries.
+        chunk_size = 25
+
+        for transactions_chunk in chunked(transaction_names, chunk_size):
+            escaped_transaction_names = [
+                transaction_name.replace('"', '\\"') for transaction_name in transactions_chunk
+            ]
+            query = " OR ".join([f'transaction:"{name}"' for name in escaped_transaction_names])
+
+            raw_results = metrics_performance.timeseries_query(
+                selected_columns=["project_id", "transaction", agg_function],
+                query=query,
+                params=params,
+                rollup=interval,
+                referrer=Referrer.API_PERFORMANCE_TRANSACTIONS_STATISTICAL_DETECTOR_STATS.value,
+                groupby=Column("transaction"),
+            )
+
+            results = {}
+            for index, datapoint in enumerate(raw_results.data or []):
+                transaction_name = datapoint["transaction"]
+                if transaction_name not in results:
+                    results[transaction_name] = {
+                        "order": index,
+                        "data": [datapoint],
+                    }
+                else:
+                    data = cast(List, results[transaction_name]["data"])
+                    data.append(datapoint)
+
+            for transaction_name, item in results.items():
+                formatted_result = SnubaTSResult(
+                    {
+                        "data": zerofill(
+                            item["data"],
+                            params["start"],
+                            params["end"],
+                            interval,
+                            "time",
+                        ),
+                        "project": project_id,
+                        "order": item["order"],
+                    },
+                    params["start"],
+                    params["end"],
+                    interval,
+                )
+                yield project_id, transaction_name, formatted_result
```

Comment:
we'd still need to format each key as `{project_id},{transaction_name}` to match what the breakpoint microservice is expecting